src/HeaderAddPlugin.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Http\Client\Plugin;
+
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Adds headers to the request only if the request does not already own it.
```

Comment:
I'm not sure about appending, I think it's an edge case, not sure, actually I have difficulties to imagine a real case.